File: src/main/java/fyordo/lifeagragator/med/base/utils/WorkspaceUtils.java
package fyordo.lifeagragator.med.base.utils;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpSession;
import org.springframework.web.context.request.RequestAttributes;
import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

public class WorkspaceUtils {
    public static Long getUserId(){
        RequestAttributes requestAttributes = RequestContextHolder
                .currentRequestAttributes();
        ServletRequestAttributes attributes = (ServletRequestAttributes) requestAttributes;
        HttpServletRequest request = attributes.getRequest();
        HttpSession httpSession = request.getSession(true);

        var attribute = httpSession.getAttribute("userId");
        if (attribute == null){
            return null;
        }

        return Long.valueOf((String)attribute);
    }
}

File: src/main/kotlin/fyordo/lifeagragator/med/base/utils/WorkspaceUtils.kt
package fyordo.lifeagragator.med.base.utils

import kotlinx.coroutines.withContext
import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes

fun getUserId(): Long? {
    val requestAttributes = RequestContextHolder
        .currentRequestAttributes()
    val attributes = requestAttributes as ServletRequestAttributes
    val request = attributes.request
    val httpSession = request.getSession(true)

    val attribute = httpSession.getAttribute("userId") ?: return null

    return (attribute as String).toLong()
}